extensions/resteasy-classic/resteasy-client/runtime/src/main/java/io/quarkus/restclient/runtime/QuarkusRestClientBuilder.java

```diff
@@ -806,6 +830,16 @@ private static BeanManager getBeanManager() {
         }
     }
 
+    private static Method[] resolveMethods(final Class<?> type) {
+        // If the type extends Closeable or AutoCloseable, we need to filter out their methods
+        if (AutoCloseable.class.isAssignableFrom(type)) {
+            return Stream.of(type.getMethods())
+                    .filter(method -> !IGNORED_METHODS.contains(method))
+                    .toArray(Method[]::new);
```

Comment:
I'm definitely no fan of this, but we can let it go in :)

Reply:
Yes, this is a copy from the original RESTEasy Builder code:
https://github.com/resteasy/resteasy-microprofile/commit/cb58ceb333396875032aeeb58d3723569edcbb6b

Initially, we used the one provided by RESTEasy, but we made a copy to remove references to their CDI integration and also to remove the CDI dependency. They diverged a bit, so I've tried to reconcile them again for consistency.